components/motor/collectors.go

```diff
@@ -47,17 +48,13 @@ func newPositionCollector(resource interface{}, params data.CollectorParams) (da
 			}
 			return nil, data.FailedToReadErr(params.ComponentName, position.String(), err)
 		}
-		return Position{Position: v}, nil
+		return pb.GetPositionResponse{
+			Position: v,
+		}, nil
 	})
 	return data.NewCollector(cFunc, params)
 }
 
-// Powered wraps the returned IsPowered value.
-type Powered struct {
-	IsPowered bool
-	PowerPct  float64
-}
-
 func newIsPoweredCollector(resource interface{}, params data.CollectorParams) (data.Collector, error) {
```

Comment:
This is moreso a question for @AaronCasas, but how did we decide which methods can have data capture configured? I see a lot of components have more methods than just the ones we have collectors registered for. For some of them, it doesn't make sense to have a collector since they don't actually return any data (for example `Stop`/`StopResponse`). But for motor for eg., `IsPowered` has a collector but `IsMoving` does not.

Reply:
No systematic way! When the original ones were added we added them for all methods that returned data. Since then, they've been added more adhoc